src/movies.js

```diff
@@ -1,13 +1,90 @@
 // Iteration 1: All directors? - Get the array of all directors.
+//console.log(movies);
+
+// function getAllDirectors(movies){
+//     return movies.map(function (movie)
+//     {
+//     return movie.director;
+// })
+// }
+
+//easier ES6 notation
+function getAllDirectors(movies){
+    return movies.map(movie => movie.director);
+}
+console.log(getAllDirectors(movies));
+console.clear();
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+// function howManyMovies(movies){
+//     let filteredMovies = movies.filter(function (movie){
+//         if (movie.director === 'Steven Spielberg') {
+//             return true
+//         }
+//         return filteredMovies;
+        
+//         }
+
+//     )}
+
+//     console.log(howManyMovies(movies));
+function howManyMovies(movies){
+
+    let filteredMovies = movies.filter(function(movie){
+        if ((movie.director === 'Steven Spielberg') && (movie.genre.includes('Drama'))){
+            return true}
+            
+    })
+    return filteredMovies.length;
+} 
```

Comment:
Filter expects a function that returns true or false, so I would explicitly return false as well. You could either add an else to the "if" and return false or you could directly return the expression since it evaluates to either true or false.

```
    let filteredMovies = movies.filter(function(movie) {
        return ((movie.director === 'Steven Spielberg') && (movie.genre.includes('Drama')));  
    })
```